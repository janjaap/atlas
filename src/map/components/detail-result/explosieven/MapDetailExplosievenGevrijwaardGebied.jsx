import React from 'react';
import PropTypes from 'prop-types';

import MapDetailResultDateItem from '../MapDetailResultDateItem';
import MapDetailResultItem from '../MapDetailResultItem';
import MapDetailResultWrapper from '../MapDetailResultWrapper';

const MapDetailExplosievenGevrijwaardGebied = ({
  panoUrl, gevrijwaardGebied, onMaximize, onPanoPreviewClick
}) => (
  <MapDetailResultWrapper
    panoUrl={panoUrl}
    onMaximize={onMaximize}
    onPanoPreviewClick={onPanoPreviewClick}
    subTitle={gevrijwaardGebied.label}
    title="Gevrijwaard gebied"
  >
    <ul className="map-detail-result__list">
      <MapDetailResultDateItem
        label="Datum rapport"
        date={gevrijwaardGebied.date}
      />
      <MapDetailResultItem
        label="Soort handeling"
        value={gevrijwaardGebied.type}
      />
      <MapDetailResultItem
        label="Bron"
        value={gevrijwaardGebied.source}
      />
      <MapDetailResultItem
        label="Opmerkingen"
        value={gevrijwaardGebied.remarks}
      />
    </ul>
  </MapDetailResultWrapper>
);

MapDetailExplosievenGevrijwaardGebied.propTypes = {
  gevrijwaardGebied: PropTypes.shape({
    date: PropTypes.instanceOf(Date),
    label: PropTypes.string,
    remarks: PropTypes.string,
    source: PropTypes.string,
    type: PropTypes.string
  }).isRequired,
  panoUrl: PropTypes.string.isRequired,
  onMaximize: PropTypes.func.isRequired,
  onPanoPreviewClick: PropTypes.func.isRequired
};

export default MapDetailExplosievenGevrijwaardGebied;
