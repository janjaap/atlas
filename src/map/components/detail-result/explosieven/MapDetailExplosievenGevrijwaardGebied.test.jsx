import React from 'react';
import { shallow } from 'enzyme';

import MapDetailExplosievenGevrijwaardGebied from './MapDetailExplosievenGevrijwaardGebied';

describe('MapDetailExplosievenGevrijwaardGebied', () => {
  it('should render everything', () => {
    const gevrijwaardGebied = {
      date: new Date('1919-12-31'),
      label: 'Gevrijwaard gebied label',
      remarks: 'Gevrijwaard gebied remarks',
      source: 'Gevrijwaard gebied source',
      type: 'Gevrijwaard gebied type'
    };
    const clickHandler = jest.fn();
    const wrapper = shallow(
      <MapDetailExplosievenGevrijwaardGebied
        panoUrl="panoUrl"
        onMaximize={clickHandler}
        onPanoPreviewClick={clickHandler}
        gevrijwaardGebied={gevrijwaardGebied}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
