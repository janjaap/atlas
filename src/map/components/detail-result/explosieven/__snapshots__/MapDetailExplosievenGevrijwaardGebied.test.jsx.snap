// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapDetailExplosievenGevrijwaardGebied should render everything 1`] = `
<MapDetailResultWrapper
  onMaximize={[MockFunction]}
  onPanoPreviewClick={[MockFunction]}
  panoUrl="panoUrl"
  subTitle="Gevrijwaard gebied label"
  title="Gevrijwaard gebied"
>
  <ul
    className="map-detail-result__list"
  >
    <MapDetailResultDateItem
      date={1919-12-31T00:00:00.000Z}
      label="Datum rapport"
    />
    <MapDetailResultItem
      label="Soort handeling"
      value="Gevrijwaard gebied type"
    />
    <MapDetailResultItem
      label="Bron"
      value="Gevrijwaard gebied source"
    />
    <MapDetailResultItem
      label="Opmerkingen"
      value="Gevrijwaard gebied remarks"
    />
  </ul>
</MapDetailResultWrapper>
`;
